ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/types/SqlNull.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.ksql.types;
+
+import io.confluent.ksql.schema.ksql.DataException;
+import io.confluent.ksql.schema.ksql.FormatOptions;
+import io.confluent.ksql.schema.ksql.SqlBaseType;
+
+public final class SqlNull extends SqlType {
```

Comment:
Hi @big-andy-coates 
I'm sure there's a good reason for it, but I am curious to understand why null is being modelled as a SqlType. I would have thought that is a value that is valid for any SqlType, not a SqlType itself. E.g. if there is null in a column, then the column does not have a type of SqlNull, it would have a type of whatever the column type is (E.g. SqlInteger) and the value would be null...